### PR TITLE
Add UI support for OCI cloud credentials and node template.

### PR DESCRIPTION
**Pull request by jlamillan**
Proposed changes
======
This PR adds support for storing Oracle Cloud Credentials and creating node-templates for the OCI [node driver](https://github.com/rancher-plugins/rancher-machine-driver-oci/).

Types of changes
======

Linked Issues
======
 
- Depends on rancher/rancher#26603

Further comments
======

### CODE REVIEW

**Comment by jlamillan** [comment]
Moving back to WIP while I move the node template code from the [rancher-plugins/ui-node-driver-oci repo](https://github.com/rancher-plugins/ui-node-driver-oci/tree/jlamillan/initial_driver_impl) to this repo (rancher/ui) so that we do not depend on an external CDN.

File: lib/nodes/addon/components/driver-oci/component.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  layout,
+
+  driverName:     'oci',
+  region:         'us-phoenix-1',
+  nodeImage:      'Oracle-Linux-7.7',
+  step:            1,
+
+  config: alias('model.ociConfig'),
+
+  init() {
```
Comment (westlywright) [inline lib/nodes/addon/components/driver-oci/component.js:63]: This isn't needed if you're not adding anything additional.

File: lib/nodes/addon/components/driver-oci/component.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    const nodeImage = get(this, 'config.nodeImage');
+
+    return nodeImage && imageMap[nodeImage];
+  }),
+  bootstrap() {
+    const config = get(this, 'globalStore').createRecord({
+      type:      'ociConfig',
+      region:    'us-phoenix-1',
+      nodeImage: 'Oracle-Linux-7.7',
+    });
```
Comment (westlywright) [inline lib/nodes/addon/components/driver-oci/component.js:139]: region and nodeImage are declared at the component level, i'd use them or get ride of them if you're only using these values here.

File: lib/nodes/addon/components/driver-oci/component.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+      return ocid.substring(start + 1, end);
+    }
+
+    // Get generic API validation errors
+    this._super();
+    var errors = get(this, 'errors') || [];
+
+    if (!get(this, 'model.name')) {
+      errors.push('Name is required');
```
Comment (westlywright) [inline lib/nodes/addon/components/driver-oci/component.js:157]: These strings should be placed in the translations file.

File: lib/nodes/addon/components/driver-oci/template.hbs [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+          placeholder=(t "nodeDriver.oci.subnet.placeholder")
+        }}
+        <p class="help-block">
+          {{t 'nodeDriver.oci.subnet.helpText'}}
+        </p>
+      </div>
+
+      <div class="col span-6">
+        <label class="acc-label">
+          Use private IP address to connect to compute instances(s)
```
Comment (westlywright) [inline lib/nodes/addon/components/driver-oci/template.hbs:161]: String should be in translations.

**Pull request merged**